packages/flutter/test/widgets/focus_manager_test.dart

```diff
@@ -745,4 +745,134 @@ void main() {
     await tester.pump();
     expect(parent1.focusedChild, equals(child2));
   });
+  testWidgets('Ancestors get notified exactly as often as needed if focused child changes focus.', (WidgetTester tester) async {
+    bool topFocus;
```

Comment:
It feels odd that in the test below not having focus is represented as this variable either being null or false. Unless there's a reason for the null/false distinction, should these just be initialized to false?

Reply:
There is a reason: `null` means that the notification callback was never called. False means that it was called, but the associated node was set to "not focused".  I wanted to test to make sure that 1) the callback was called, and 2) that it was called with the expected value.  In addition, I wanted to make sure that it was not called more than once per `pump`, so I added the `*Notify` counts.

I suppose I could initialize this to false, and just use the notification count to make sure it was called, if that's clearer.

Reply:
Oh, yeah, I thought that's what the counts were for... I think that would make this a little clearer.

Reply:
OK, I set the bools all to false, and rely on the counts to make sure it was called the right number of times.